eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -55,17 +54,19 @@
 public final class DiscoverMojo extends SafeMojo {
 
     @Override
-    public void exec() throws IOException {
+    public void exec() throws FileNotFoundException {
         final Collection<ForeignTojo> tojos = this.scopedTojos().notDiscovered();
         final Collection<String> discovered = new HashSet<>(1);
         for (final ForeignTojo tojo : tojos) {
             final Path src = tojo.optimized();
-            final Collection<String> names = this.discover(src);
-            for (final String name : names) {
-                this.scopedTojos().add(name).withDiscoveredAt(src);
-                discovered.add(name);
-            }
-            tojo.withDiscovered(names.size());
+            tojo.withDiscovered(
```

Comment:
@volodya-lombrozo what's the reason to turn these lines into stream pipeline? Is it about speed optimization or readability?

Reply:
@maxonfjvipon It's is about code [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity):
> Some studies[[12]](https://en.wikipedia.org/wiki/Cyclomatic_complexity#cite_note-schroeder99-14) find a positive correlation between cyclomatic complexity and defects: functions and methods that have the highest complexity tend to also contain the most defects

Before it was 2 nested loops, now it's just a single loop

Reply:
@maxonfjvipon It's is about code [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity):
> Some studies[[12]](https://en.wikipedia.org/wiki/Cyclomatic_complexity#cite_note-schroeder99-14) find a positive correlation between cyclomatic complexity and defects: functions and methods that have the highest complexity tend to also contain the most defects

Before it was 2 nested loops, now it's just a single loop

Reply:
@volodya-lombrozo fair enough